framework/include/auxkernels/MaterialAuxBase.h

```diff
@@ -54,17 +45,18 @@ class MaterialAuxBase : public AuxKernel
    */
   virtual ~MaterialAuxBase(){}
 
-protected:
+  virtual Real computeValue();
 
+protected:
+  virtual Real computePropertyValue() = 0;
   /// Reference to the material property for this AuxKernel
   const MaterialProperty<T> & _prop;
 
+private:
   // Value to be added to the material property
   const Real _factor;
-
   // Multiplier for the material property
```

Comment:
Can you please change these ```//``` into ```///``` while you are working on this. Otherwise the comments won't get picked up by doxygen.